tools/analysis/analysis_test.go

```diff
@@ -259,6 +256,190 @@ func TestCheckError(t *testing.T) {
+			importingLocation common.Location,
+			importRange ast.Range,
+		) ([]byte, error) {
+			switch location {
+			case contractLocation:
+				return []byte(contractCode), nil
+			default:
+				require.FailNowf(t,
+					"import of unknown location",
+					"location: %s",
+					location,
+				)
+				return nil, nil
+			}
+		},
+		HandleCheckerError: func(err analysis.ParsingCheckingError, _ *sema.Checker) error {
+			require.Error(t, err)
+			handlerCalls++
+			return nil
+		},
+	}
+
+	require.Equal(t, 1, handlerCalls)
+
+	programs, err := analysis.Load(config, contractLocation)
+	require.NoError(t, err)
+
+	var checkerError *sema.CheckerError
+	require.ErrorAs(t, programs[contractLocation].LoadError, &checkerError)
+}
+
+// Tests that an error handled by the custom error handler is not returned
+// However, it must set LoadError to the handled error so that checkers later importing the program can see it
+func TestHandledLoadErrorImportedProgram(t *testing.T) {
+
+	t.Parallel()
+
+	contract1Address := common.MustBytesToAddress([]byte{0x1})
+	contract1Location := common.AddressLocation{
+		Address: contract1Address,
+		Name:    "ContractA",
+	}
+	const contract1Code = `
+	  import ContractB from 0x2
+	  
+      access(all) contract ContractA {
+	    init() {}
+	  }
+	`
+	contract2Address := common.MustBytesToAddress([]byte{0x2})
+	contract2Location := common.AddressLocation{
+		Address: contract2Address,
+		Name:    "ContractB",
+	}
+	const contract2Code = `
+	  access(all) contract ContractB {
+	    init() {
+	      X
+	    }
+	  }
+	`
+
+	handlerCalls := 0
+	config := &analysis.Config{
+		Mode: analysis.NeedTypes,
+		ResolveCode: func(
+			location common.Location,
+			importingLocation common.Location,
+			importRange ast.Range,
+		) ([]byte, error) {
+			switch location {
+			case contract1Location:
+				return []byte(contract1Code), nil
+			case contract2Location:
+				return []byte(contract2Code), nil
+			default:
+				require.FailNowf(t,
+					"import of unknown location",
+					"location: %s",
+					location,
+				)
+				return nil, nil
+			}
+		},
+		HandleCheckerError: func(err analysis.ParsingCheckingError, _ *sema.Checker) error {
+			require.Error(t, err)
+			handlerCalls++
+			return nil
+		},
+	}
+
+	require.Equal(t, 2, handlerCalls)
+
+	programs, err := analysis.Load(config, contract1Location)
+	require.NoError(t, err)
+
+	var checkerError *sema.CheckerError
+	require.ErrorAs(t, programs[contract1Location].LoadError, &checkerError)
+	require.ErrorAs(t, programs[contract2Location].LoadError, &checkerError)
+
+	// Validate that parent checker receives the imported program error despite it being handled
+	var importedProgramErr *sema.ImportedProgramError
+	loadErr := programs[contract1Location].LoadError.(analysis.ParsingCheckingError)
+	unwrapedErr := loadErr.Unwrap().(*sema.CheckerError)
+	require.Len(t, unwrapedErr.ChildErrors(), 1)
+	require.ErrorAs(t, unwrapedErr.ChildErrors()[0], &importedProgramErr)
```

Comment:
`ErrorAs` does all this unwrapping, so all that is needed here is: 

```suggestion
	require.ErrorAs(t, programs[contract1Location].LoadError, &importedProgramErr)
```